modules/framework/galasa-parent/dev.galasa.framework.api.streams/src/test/java/dev/galasa/framework/api/streams/internal/routes/StreamsRouteTest.java

```diff
@@ -0,0 +1,155 @@
+        MockHttpServletResponse servletResponse = new MockHttpServletResponse();
+        ServletOutputStream outStream = servletResponse.getOutputStream();
+
+        // When...
+        mockServlet.init();
+        mockServlet.doGet(mockRequest, servletResponse);
+
+        String output = outStream.toString();
+
+        assertThat(servletResponse.getStatus()).isEqualTo(200);
+        assertThat(servletResponse.getContentType()).isEqualTo("application/json");
+        assertThat(getJsonArrayFromJson(output, "streams")).hasSize(1);
+    }
+
+    @Test
+    public void testGetStreamsRouteReturnsStreamsArrayWithMultipleStreamsOK() throws Exception {
+
+        // Given...
+        Map<String, String> headerMap = Map.of("Authorization", "Bearer " + BaseServletTest.DUMMY_JWT);
+
+        List<IStream> mockStreams = new ArrayList<>();
+        MockStream mockstream = new MockStream("testStream", "This is a dummy test stream",
+                "http://mymavenrepo.host/testmaterial",
+                "http://mymavenrepo.host/testmaterial/com.ibm.zosadk.k8s/com.ibm.zosadk.k8s.obr/0.1.0-SNAPSHOT/testcatalog.yaml",
+                "fake-obr-location", true);
+
+        MockStream mockstream2 = new MockStream("testStream2", "This is a second dummy test stream",
+                "http://mymavenrepo.host/testmaterialdummy",
+                "http://mymavenrepo.host/testmaterialdummy/com.ibm.zosadk.k8s/com.ibm.zosadk.k8s.obr/0.1.0-SNAPSHOT/testcatalog.yaml",
+                "fake-obr-location", true);
+
+        mockStreams.add(mockstream);
+        mockStreams.add(mockstream2);
+
+        MockIStreamsService mockIStreamsService = new MockIStreamsService(mockStreams);
+        MockRBACService mockRBACService = FilledMockRBACService.createTestRBACServiceWithTestUser(JWT_USERNAME);
+        MockFramework mockFramework = new MockFramework(mockRBACService, mockIStreamsService);
+        MockIConfigurationPropertyStoreService mockIConfigurationPropertyStoreService = new MockIConfigurationPropertyStoreService(
+                "framework");
+
+        MockEnvironment env = new MockEnvironment();
+        env.setenv(EnvironmentVariables.GALASA_USERNAME_CLAIMS, "preferred_username");
+
+        MockStreamsServlet mockServlet = new MockStreamsServlet(mockFramework, env,
+                mockIConfigurationPropertyStoreService);
+
+        MockHttpServletRequest mockRequest = new MockHttpServletRequest(null, headerMap);
+        MockHttpServletResponse servletResponse = new MockHttpServletResponse();
+        ServletOutputStream outStream = servletResponse.getOutputStream();
+
+        // When...
+        mockServlet.init();
+        mockServlet.doGet(mockRequest, servletResponse);
+
+        String output = outStream.toString();
+
+        assertThat(servletResponse.getStatus()).isEqualTo(200);
+        assertThat(servletResponse.getContentType()).isEqualTo("application/json");
+        assertThat(getJsonArrayFromJson(output, "streams")).hasSize(2);
+    }
```

Comment:
Maybe it's also worth adding a test to check the response if the streams service fails to get the streams?